src/main/kotlin/com/exactpro/th2/codec/Application.kt

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright 2023 Exactpro (Exactpro Systems Limited)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.exactpro.th2.codec
+
+import com.exactpro.th2.codec.configuration.ApplicationContext
+import com.exactpro.th2.codec.configuration.Configuration
+import com.exactpro.th2.codec.configuration.TransportType.PROTOBUF
+import com.exactpro.th2.codec.configuration.TransportType.TH2_TRANSPORT
+import com.exactpro.th2.codec.proto.ProtoDecodeProcessor
+import com.exactpro.th2.codec.proto.ProtoDecoder
+import com.exactpro.th2.codec.proto.ProtoEncodeProcessor
+import com.exactpro.th2.codec.proto.ProtoEncoder
+import com.exactpro.th2.codec.transport.TransportDecodeProcessor
+import com.exactpro.th2.codec.transport.TransportDecoder
+import com.exactpro.th2.codec.transport.TransportEncodeProcessor
+import com.exactpro.th2.codec.transport.TransportEncoder
+import com.exactpro.th2.common.schema.factory.CommonFactory
+import mu.KotlinLogging
+
+class Application(commonFactory: CommonFactory) : AutoCloseable {
+
+    private val configuration = Configuration.create(commonFactory)
+    private val context = ApplicationContext.create(configuration, commonFactory)
+    private val protoRouter = context.commonFactory.messageRouterMessageGroupBatch
+    private val transportRouter = context.commonFactory.transportGroupBatchRouter
+
+    private val codecs: List<AutoCloseable> = mutableListOf<AutoCloseable>().apply {
+        configuration.transportLines.forEach { (prefix, type) ->
+            when (type) {
+                PROTOBUF -> {
+                    createProtoDecoder("${prefix}_decoder", "${prefix}_decoder_in", "${prefix}_decoder_out")
+                    createProtoEncoder("${prefix}_encoder", "${prefix}_encoder_in", "${prefix}_encoder_out")
+                }
+
+                TH2_TRANSPORT -> {
+                    createTransportDecoder("${prefix}_decoder", "${prefix}_decoder_in", "${prefix}_decoder_out")
+                    createTransportEncoder("${prefix}_encoder", "${prefix}_encoder_in", "${prefix}_encoder_out")
+                }
```

Comment:
Maybe do something like this instead?
```suggestion
            val createEncoder = when (type) {
                PROTOBUF -> this::createProtoEncoder
                TH2_TRANSPORT -> this::createTransportEncoder
            }
            createEncoder("${prefix}_encoder", "${prefix}_encoder_in", "${prefix}_encoder_out")
            val createDecoder = when (type) {
                PROTOBUF -> this::createProtoDecoder
                TH2_TRANSPORT -> this::createTransportDecoder
            }
            createDecoder("${prefix}_decoder", "${prefix}_decoder_in", "${prefix}_decoder_out")
```